tests/basic/shared_nfsv4_test.go

```diff
@@ -0,0 +1,173 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/portworx/torpedo/tests"
+)
+
+const (
+	defaultCommandRetry   = 5 * time.Second
+	defaultCommandTimeout = 1 * time.Minute
+)
+
+var _ = Describe("{NFSServerFailover}", func() {
+	var contexts []*scheduler.Context
+
+	It("has to setup, validate, failover, make sure pods on old server got restarted, and teardown apps", func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nfsserverfailover-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+
+		for _, ctx := range contexts {
+			oneMinute := time.Second * time.Duration(60)
+			var nodesNonReplica []node.Node
+			var volume *volume.Volume
+			Step("disable scheduling on non replica nodes", func() {
+				vols, err := Inst().S.GetVolumes(ctx)
+				if err != nil {
+					logrus.Infof("get volumes error %v", err)
+				}
+				volume = vols[0]
+				replicaSets, err := Inst().V.GetReplicaSets(volume)
+				if err != nil {
+					logrus.Infof("get replicaSets error %v", err)
+				}
+				var replicaNodes []string
+				for _, replicaSet := range replicaSets {
+					nodes := replicaSet.Nodes
+					replicaNodes = append(replicaNodes, nodes...)
+				}
+				allNodes := node.GetWorkerNodes()
+				for _, node := range allNodes {
+					if !contains(replicaNodes, node.VolDriverNodeID) {
```

Comment:
We could use map instead of slice for `replicaNodes` and get rid of the `contains` function.